Formula/oras.rb

```diff
@@ -0,0 +1,24 @@
+class Oras < Formula
+  desc "OCI Registry As Storage"
+  homepage "https://github.com/deislabs/oras"
+  url "https://github.com/deislabs/oras/archive/v0.10.0.tar.gz"
+  sha256 "ace01c2e484ae91f88ecf338daf3a023692acf4e4814c33887f1a66d5a49bb3d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/deislabs/oras/internal/version.Version=#{version}
+      -X github.com/deislabs/oras/internal/version.BuildMetadata=Homebrew
+    ]
+    system "go", "build", *std_go_args,
+                          "-ldflags", ldflags.join(" "),
+                          "./cmd/oras"
+  end
+
+  test do
+    assert_match "#{version}+Homebrew", shell_output("#{bin}/oras version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- _**If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.**_
- _**Same if the software requires a virtual machine, docker instance, etc. to be running.**_

😉

Reply:
It's been so long since I've reviewed a new formula that I forgot about the "fails as expected" case.